elastic/elastic-stack-testing
# Test parity of Kibana monitoring docs

Resolves #25.

The goal of this PR is to:
1. Spin up Elasticsearch with X-Pack (i.e. default build).
2. Let Kibana do its internal monitoring.
3. Collect resulting docs from `.monitoring-kibana-*` as samples.
4. Turn off Kibana's internal monitoring and turn on Metricbeat monitoring Kibana instead.
5. Collect resulting docs from `.monitoring-kibana-mb-*` as samples.
6. Compare the two samples and verify that they are structurally identical (except expected differences).

Note that, for now, this PR and the code within it narrowly focusses specifically on Kibana monitoring documents. I'm deliberately not trying to make it generic just yet. Once we have the need to implement a similar parity check for other products, I'll refactor the code as necessary then.

### Testing this PR

1. Check out this PR and switch to the base folder of this repo in your local checkout.
2. Edit `playbooks/monitoring/buildenv.sh` and set `ES_BUILD_URL` to `snapshots.elastic.co/6.5.0-xxxxxxxx` where `xxxxxxxx` is a hash that can be found from the console output of the latest successful build of https://internal-ci.elastic.co/view/Release/job/elastic+release-manager+6.x+unified-snapshot/.
3. Run `./playbooks/monitoring/buildenv.sh`. If you don't have pre-requisites already setup, see https://github.com/elastic/elastic-stack-testing#environment-setup.
4. The script should exit (after about 5 minutes) with exit code 0. STDOUT should say things like:
   ```
   OK: Metricbeat-indexed doc for type='kibana_stats' has expected parity with kibana-indexed doc.
   OK: Metricbeat-indexed doc for type='kibana_settings' has expected parity with kibana-indexed doc.
   ```
